### PR TITLE
Search UI Revamp, Assets Addition, Code Clean Up and Theme Bug Fixes

### PR DESCRIPTION
**Pull request by akshaaatt**
- Removed SearchFragment.kt
- Added Vector Assets
- Cleaned Up the Code at places
- Fixed the Onboarding theme
- Icon Attributions have been added

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -23,8 +23,14 @@ android {
 
     buildTypes {
         release {
-            minifyEnabled false
-            // proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            minifyEnabled true
```
Comment (amCap1712) [inline app/build.gradle:26]: There are a few issues with minification. We want to add the ability for users to submit stack traces in future. Minifying reduces the usability of such stack traces. Of course, we can add some settings to reach a middle ground but that'll take some extra work. Also, the application size is small enough that size is not much of an issue. I would prefer to leave this out of this PR and discuss this later.

File: app/src/main/assets/about.html [hunk 2]
```diff
@@ -11,4 +11,8 @@ <h5>Development</h5>
 <h5>Bugs</h5>
 <p>
 You should be aware that error logs are sent from this app in the event of a crash. Personally identifiable information is not included. You can disable this in the settings menu if you like.
+</p>
+<h5>Attributions</h5>
+<p>
+This app uses Icons made by Freepik, Pixel Perfect, Good Ware, photo3idea_studio and Pixelmeetup from www.flaticon.com
```
Comment (amCap1712) [inline app/src/main/assets/about.html:17]: Freepik seems to prefer the word [designed](https://support.freepik.com/hc/en-us/articles/208976585-Attribution-How-when-and-where-). Let's use that. While we are at it, let's also add a link to Freepik.

**Review by amCap1712 — CHANGES_REQUESTED** [review]
A couple of comments to be resolved. Overall changes look good. Thanks!

File: app/build.gradle (hunk 1 above)
Comment (akshaaatt) [reply app/build.gradle]: Makes sense. Done.

File: app/src/main/assets/about.html (hunk 2 above)
Comment (akshaaatt) [reply app/src/main/assets/about.html]: Added designed and the link to flaticon was added at the end of the sentence.

**Comment by akshaaatt** [comment]
Done :)

File: app/build.gradle [hunk 3]
```diff
@@ -24,7 +24,11 @@ android {
     buildTypes {
         release {
             minifyEnabled false
-            // proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
Comment (amCap1712) [inline app/build.gradle:27]: This looks wrong? We only need the release block once methinks.

**Review by amCap1712 — CHANGES_REQUESTED** [review]

**Review by amCap1712 — APPROVED** [review]

**Pull request merged**